### ISSUE TITLE
Support j2sdk 5.0

### ISSUE THREAD

**Issue by sourceforge-issue-exporter** [beanshell/beanshell#99]
Support j2sdk 5.0 futures like enum, foreach ....

Reported by: majo44

Original Ticket: "beanshell/feature-requests/10":https://sourceforge.net/p/beanshell/feature-requests/10

**Comment by nickl-** [comment]
foreach:

```java
bsh % for (a: new int[]{1,2,3}) print(a);
1I :Integer
2I :Integer
3I :Integer
--> void
```

**Comment by nickl-** [comment]
enum parseable:

```java
bsh % enum Level {
LOW,
MEDIUM,
HIGH
}
--> $3 = class Level :Class
```

**Comment by nickl-** [comment]
NPE on enum:

```java
bsh % Level.LOW;
// Error: Unknown error: java.lang.NullPointerException: Cannot invoke "bsh.BshClassManager$MemberCache.findField(String)" because the return value of "bsh.util.ReferenceCache.get(Object)" is null
```

**Comment by nickl-** [comment]
Enum implemented as per #80 

Closed implemented
